app/controllers/areas_controller.rb

```diff
@@ -18,36 +21,27 @@ def edit
 
   def create
     @area = Area.new(area_params)
-    respond_to do |format|
-      if @area.save
-        format.html { redirect_to @area, notice: 'Area was successfully created.' }
-        format.json { render :show, status: :created, location: @area }
-      else
-        format.html { render :new }
-        format.json { render json: @area.errors, status: :unprocessable_entity }
-      end
-    end
+    @area.city_id=params[:city_id]
+   if @area.save
+     @areas = @city.areas.all
```

Comment:
why do need all area objects in create ? can you use ajax requests for this(assuming you are using this for select dropdown)